packages/std/src/math/decimal.rs

```diff
@@ -553,6 +567,40 @@ mod tests {
         assert_eq!(left * right, Uint128(0));
     }
 
+    #[test]
+    fn decimal_uint128_division() {
+        // a/b
+        let left = Decimal::percent(150); // 1.5
+        let right = Uint128(3);
+        assert_eq!(left / right, Decimal::percent(50));
+
+        // 0/a
+        let left = Decimal::zero();
+        let right = Uint128(300);
+        assert_eq!(left / right, Decimal::zero());
+    }
+
+    #[test]
+    #[should_panic(expected = "attempt to divide by zero")]
+    fn decimal_uint128_divide_by_zero() {
+        let left = Decimal::percent(150); // 1.5
+        let right = Uint128(0);
+        let _result = left / right;
+    }
+
+    #[test]
+    fn decimal_uint128_div_assign() {
+        // a/b
+        let mut dec = Decimal::percent(150); // 1.5
+        dec /= Uint128(3);
+        assert_eq!(dec, Decimal::percent(50));
+
+        // 0/a
+        let mut dec = Decimal::zero();
+        dec /= Uint128(300);
+        assert_eq!(dec, Decimal::zero());
+    }
```

Comment:
Could you create a panic test here as well, especially since this is not implemented via `ops::Div`?